src/tests/simulation/tests/integration_tests/recovery/event_log.rs

```diff
@@ -0,0 +1,176 @@
+        file.write_all(
+            "\
+[server]
+telemetry_enabled = false
+metrics_level = \"Disabled\"
+[meta]
+event_log_flush_interval_ms = 10\
+        "
+            .as_bytes(),
+        )
+        .expect("failed to write config file");
+        file.into_temp_path()
+    };
+
+    Configuration {
+        config_path: ConfigPath::Temp(config_path.into()),
+        frontend_nodes: 1,
+        compute_nodes: 1,
+        meta_nodes: 1,
+        compactor_nodes: 0,
+        compute_node_cores: 2,
+        etcd_timeout_rate: 0.0,
+        etcd_data_path: None,
+    }
+}
+
+async fn assert_event_count(mut session: Session, expected_count: u64) {
+    let event_type = risingwave_pb::meta::event_log::EventType::CreateStreamJobFail.as_str_name();
+    let count = session
+        .run(format!(
+            "select count(*) from rw_event_logs where event_type='{}'",
+            event_type
+        ))
+        .await
+        .unwrap();
+    assert_eq!(count, expected_count.to_string());
+}
+
+async fn assert_latest_event(mut session: Session, name: impl ToString, definition: impl ToString) {
+    let event_type = risingwave_pb::meta::event_log::EventType::CreateStreamJobFail.as_str_name();
+    let info = session
+        .run(format!(
+            "select info from rw_event_logs where event_type='{}' order by timestamp desc limit 1",
+            event_type
+        ))
+        .await
+        .unwrap();
+    let json = serde_json::from_str::<serde_json::Value>(&info).unwrap();
+    // the event log shows the detail of the failed creation.
+    assert_eq!(
+        json.get("name").unwrap().as_str().unwrap(),
+        name.to_string()
+    );
+    assert_eq!(
+        json.get("definition")
+            .unwrap()
+            .as_str()
+            .unwrap()
+            .to_lowercase(),
+        definition.to_string().to_lowercase()
+    );
+}
+
+/// Tests event log do record info of stream job creation failure, for CREATE TABLE/MV/INDEX/SINK.
+#[tokio::test]
+async fn test_create_stream_job_fail() -> Result<()> {
+    let mut cluster = Cluster::start(cluster_config()).await.unwrap();
+    let mut session = cluster.start_session();
+    const WAIT_RECOVERY_SEC: u64 = 5;
+    let mut expect_failure_count = 0;
+
+    // create table succeeds.
+    session
+        .run("create table t1 (c int primary key);")
+        .await
+        .unwrap();
+    assert_event_count(session.clone(), expect_failure_count).await;
+    // create table fails due to injected failure and subsequent recovery.
+    let fp_start_actors_err = "start_actors_err";
+    let create_should_fail = "create table t2 (c int primary key)";
+    fail::cfg(fp_start_actors_err, "return").unwrap();
+    session.run(create_should_fail).await.unwrap_err();
+    fail::remove(fp_start_actors_err);
+    expect_failure_count += 1;
+    assert_event_count(session.clone(), expect_failure_count).await;
+    assert_latest_event(session.clone(), "t2", create_should_fail).await;
```

Comment:
Can we wrap this is some test function to make it more readable?

Reply:
Fixed.